gui/browserv7/src/RBrowser.cxx

```diff
@@ -593,6 +593,15 @@ void RBrowser::Hide()
       fWebWindow->CloseConnections();
 }
 
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+/// Run widget Sync method - processing pending actions
+
+void RBrowser::Sync()
```

Comment:
When is this method called?

Reply:
I put it here for future use.
If one want to handle timeouts with connection in RBrowser,
one should regularly call this function

Reply:
So in the case of rootbrowse for example, would I have to call this method in the [busy loop at the end](https://github.com/root-project/root/blob/master/main/src/rootbrowse.cxx#L165)?
Or should that be done by `RWebBrowserImp`? (e.g. in `Refresh`  - which I would then need to call probably)

Reply:
It's a bit unfortunate that we have to do this polling though - this was supposed to be a signal/callback (otherwise we might as well have a `IsConnectionOpen()` method that we can query, rather than `Sync()` which goes through additional hoops).
We could probably avoid that if we had a way to set this ClearOnClose object onto the Browser itself, no?

Reply:
Yes, we can provide extra method to check connections - and invoke `Sync()` inside this method.